users/models.py

```diff
@@ -0,0 +1,69 @@
+from django.db import models
+from django.db.models.signals import pre_save
+from django.core.exceptions import ValidationError
+from django.core.validators import validate_email
+from django.utils.translation import gettext_lazy as _
+from django.contrib.auth.models import AbstractUser
+
+
+def validate_email_addr(email):
+    try:
+        validate_email(email)
+    except ValidationError:
+        raise ValidationError('Email is not valid')
+
+
+def validate_phone_number(phone_number):
+    if phone_number is not None:
+        if phone_number[0] == '+':
+            if phone_number[:4] != '+972':
+                raise ValidationError(
+                    _(f'{phone_number} is not supported'),
+                    params={'phone_number': phone_number},
+                )
+            if len(phone_number) != 13:
+                raise ValidationError(
+                    _(f'{phone_number} is not a valid phone number. The number should have 13 digits'),
+                    params={'phone_number': phone_number},
+                )
+            phone_number = phone_number[1:]
+        else:
+            if phone_number[:2] != '05':
+                raise ValidationError(
+                    _(f'{phone_number} is not a valid phone number. The number should start with 05'),
+                    params={'phone_number': phone_number},
+                )
+            if len(phone_number) != 10:
+                raise ValidationError(
+                    _(f'{phone_number} is not a valid phone number. The number should have 10 digits'),
+                    params={'phone_number': phone_number},
+                )
+        if not phone_number.isnumeric():
+            raise ValidationError(
+                _(f'{phone_number} has at least one character'),
+                params={'phone_number': phone_number},
+            )
```

Comment:
the place of the condition includes both cases
1)+972
2) 05***
but the "+" is not numeric which means that you will get a validation error each time you try to register
maybe use this condition inside the "else" statement
or remove the first character before running this condition